src/adapter/threads.ts

```diff
@@ -398,7 +398,7 @@ export class Thread implements VariableStoreDelegate {
       }
 
       this._pausedDetails = this._createPausedDetails(event);
-      this._pausedDetails[kPausedEventSymbol] = event;
```

Comment:
Not sure how much refactoring you wanna do here, but this is a good use case for a `WeakMap` (and actually most WeakMap polyfills work almost the same way here--defining a non-enumerable symbol on the key objects 😛 )